android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -274,12 +274,85 @@ protected void triggerLayoutInflation() {
         }
     }
 
+    private class BuySendSwapUiInfo {
+        public boolean shouldShowBuyControls;
+        public String titleText;
+        public String secondText;
+        public boolean shouldDisableButton;
+        public String buttonText;
+        public String linkUrl;
+    }
+
+    private BuySendSwapUiInfo getPerNetworkUiInfo(String chainId) {
+        BuySendSwapUiInfo buySendSwapUiInfo = new BuySendSwapUiInfo();
+
+        if (chainId.equals(BraveWalletConstants.MAINNET_CHAIN_ID)) {
+            buySendSwapUiInfo.shouldShowBuyControls = true;
+            buySendSwapUiInfo.buttonText = getString(R.string.wallet_buy_mainnet_button_text);
+            return buySendSwapUiInfo;
+        }
+
+        buySendSwapUiInfo.shouldShowBuyControls = false;
+        buySendSwapUiInfo.titleText = getString(R.string.wallet_test_faucet_title);
+        buySendSwapUiInfo.buttonText = getString(R.string.wallet_test_faucet_button_text);
+
+        buySendSwapUiInfo.secondText = getString(R.string.wallet_test_faucet_second_text);
+        buySendSwapUiInfo.secondText = String.format(
+                buySendSwapUiInfo.secondText, Utils.getNetworkShortText(this, chainId));
+
+        switch (chainId) {
+            case BraveWalletConstants.RINKEBY_CHAIN_ID:
+                buySendSwapUiInfo.linkUrl = "https://www.rinkeby.io/#stats";
```

Comment:
maybe let's put all const in Utils.java for now? Ideally it would be nice to pull that from mojo, perhaps after MVP

Reply:
sure, I will move that :+1:

Reply:
fixed at https://github.com/brave/brave-core/compare/24bafdf20e35480ff29fe8760671d9b11d17ad92..e067befb1f30b2731ef128e7f365570cf36c6674